godtools/App/Share/Data/UserAuthentication/AuthenticationProviders/AppleAuthentication+AuthenticationProviderInterface.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  AppleAuthentication+AuthenticationProviderInterface.swift
+//  godtools
+//
+//  Created by Rachael Skeath on 5/16/23.
+//  Copyright © 2023 Cru. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import SocialAuthentication
+import Combine
+
+extension AppleAuthentication: AuthenticationProviderInterface {
+    func getPersistedAccessToken() -> AuthenticationProviderAccessToken? {
+        
+        // TODO: - access token renewal will come through MobileContentAPI?
+        
+        return nil
+    }
+    
+    func authenticatePublisher(presentingViewController: UIViewController) -> AnyPublisher<AuthenticationProviderAccessToken?, Error> {
+        
+        return authenticatePublisher()
+            .map { (response: AppleAuthenticationResponse) in
+                
+                guard
+                    let idToken = response.identityToken,
+                    let givenName = response.fullName?.givenName,
+                    let familyName = response.fullName?.familyName
```

Comment:
Hey @rachaelblue just thought of something here as well.  I'm wondering if we ever run into a situation where we get an identity token, but given name or family name is null.  I'm good with passing this for now, but will need to make a ticket and do some research here and some testing as well to see if we can pass a null givenName or familyName to the mobile-content-api auth.